### ISSUE TITLE
CodePipelineEvent

### ISSUE THREAD

**Issue by abadger2k11** [aws/aws-lambda-java-libs#16]
Please consider adding CodePipelineEvent to aws-lambda-java-events.

**Comment by knizami** [comment]
+1 - we were unable to define our own.  @abadger2k11 were you able to define your own event that worked?

**Comment by abadger2k11** [comment]
Sort of. I hacked together a class that uses the Jackson library (com.fasterxml.jackson, for json parsing), Java zip utilities (java.util.zip.ZipEntry, java.util.zip.ZipInputStream, for unzipping CodePipeline artifacts), S3 client (com.amazonaws.services.s3.AmazonS3ClientBuilder, to fetch CodePipeline artifacts), and CodePipeline client (com.amazonaws.services.codepipeline.AWSCodePipelineClientBuilder, to report job status back to CodePipeline), and then pass it the json that CodePipeline sends in the Lambda invocation.

**Comment by asuresh8** [comment]
Yeah we couldn't really figure out a correct Java model for the code pipeline events so that's why we didn't do it with the v2 events package. We can prioritize it for v3, or if you come up with a model that works, we can add it in to the package.

**Comment by zhangzhx** [comment]
+1 I want to embed Lambda to CodePipeline as well.

**Comment by brandon-garcia** [comment]
+1 this is something I'd like to see as well, or at least a java example of how it could be done.

**Comment by raupachz** [comment]
+1

**Comment by dan-thomson-bbc** [comment]
As a workaround you can create your own CodePipelineJobDetails class with represents: https://docs.aws.amazon.com/codepipeline/latest/userguide/action-reference-Lambda.html#action-reference-Lambda-event

The use Map<String, CodePipelineJobDetails> as the input type to handleRequest. 
``` public String handleRequest(Map<String,CodePipelineJobDetails> input, Context context) {```
You will need to retrieve the details out the map as Java can't have full-stop characters in the field: 

```CodePipelineJobDetails codePipelineJobDetails = input.get("CodePipeline.job");```

**Comment by msailes** [comment]
Just to confirm.

https://docs.aws.amazon.com/lambda/latest/dg/services-codepipeline.html

Is this the integration you're interested in?

Thanks

**Comment by abadger2k11** [comment]
Yes. For my original problem when I opened this I wanted a model that would consume that CodePipeline event and give me easy access to the artifacts referenced in it. I was trying to use a Lambda function to perform a build process. For context, note that I originally opened this issue prior to the release of CodeBuild. So in my case my problem was largely solved a different way by the release of CodeBuild.

**Comment by cristianocca** [comment]
Any updates on this? Can https://github.com/aws/aws-lambda-java-libs/pull/363/files be included in v3?
